src/services/tracing.ts

```diff
@@ -171,6 +172,74 @@ export const asyncRunInSpan = <F extends () => ReturnType<F>>(
   return ret
 }
 
+// eslint-disable-next-line  @typescript-eslint/no-explicit-any
+export const wrapToRunInSpan = <A extends Array<any>, R>({
+  fn,
+  namespace,
+  spanAttributes = {},
+}: {
+  fn: (...args: A) => R
+  namespace: string
+  spanAttributes?: SpanAttributes
+}) => {
+  return (...args: A): R => {
+    const name = fn.name
+    const spanName = `${namespace}.${name}`
+    const attributes = { [SemanticAttributes.CODE_FUNCTION]: name, ...spanAttributes }
+    if (args && args.length > 0) {
+      let params = { "0": args[0] }
+      if (typeof args[0] === "object") params = args[0]
+      for (const key in params) {
+        attributes[`${SemanticAttributes.CODE_FUNCTION}.params.${key}`] = params[key]
+      }
+    }
+    const ret = tracer.startActiveSpan(spanName, { attributes }, (span) => {
+      const ret = fn(...args)
+      if (ret instanceof Error) {
+        span.recordException(ret)
+      }
+      span.end()
+      return ret
+    })
+    return ret
+  }
+}
+
+type PromiseReturnType<T> = T extends Promise<infer Return> ? Return : T
+
+// eslint-disable-next-line  @typescript-eslint/no-explicit-any
+export const wrapAsyncToRunInSpan = <A extends Array<any>, R>({
+  fn,
+  namespace,
+  spanAttributes = {},
+}: {
+  fn: (...args: A) => Promise<PromiseReturnType<R>>
+  namespace: string
+  spanAttributes?: SpanAttributes
+}) => {
+  return (...args: A): Promise<PromiseReturnType<R>> => {
+    const name = fn.name
```

Comment:
This section that sets up the span attributes is duplicate with `wrapToRunInSpan` - could probably extract into a helper function or?

Reply:
:) in general I prefer a helper but in a previous PR a rule was defined against this if the code is not used more than twice

Reply:
done https://github.com/GaloyMoney/galoy/pull/837/commits/fea23b3ea8b3294e33d4d6cbb202b35aac28de88